lib/rex/ui/text/dispatcher_shell.rb

```diff
@@ -210,6 +210,31 @@ def cmd_help(cmd=nil, *ignored)
           print(File.read(File.join(docs_dir + '.md')))
         end
       end
+
+      # smb, mysql, mssql and postgresql session types make use of `shell.session` which type can then
+      # be called against (`shell.session.type`)
+      #
+      # Meterpreter seems to make use of `shell.client` which type can then be called against (`shell.client.type`)
+      #
+      # checking this to not raise a no methods error
+      unless defined?(shell.session).nil?
```

Comment:
Is this the same as:

```suggestion
      if defined?(shell.session)
```